File: algo/240. Search a 2D Matrix II.js
/**
 * @param {number[][]} matrix
 * @param {number} target
 * @return {boolean}
 */
var searchMatrix = function(matrix, target) {
    if (matrix.length == 0 || matrix[0].length == 0) {
        return false;
    }
    let m = 0;
    let n = matrix[0].length-1;
    while (m < matrix.length && n >= 0) {
        if (matrix[m][n] == target) {
            return true;
        } else if (matrix[m][n] < target) {
            m += 1;
        } else {
            n -= 1;
        }
    }
    return false;
};

File: round3/240. Search a 2D Matrix II.cs
public class Solution {
    public bool SearchMatrix(int[][] matrix, int target) {
        int m = matrix.Length, n = matrix[0].Length;
        int i = m - 1, j = 0;
        while (i >= 0 && j < n) {
            if (matrix[i][j] == target) return true;
            if (matrix[i][j] < target) j ++;
            else if (matrix[i][j] > target) i --;
        }
        return false;
    }
}